docs-client/build.gradle

```
@@ -28,9 +30,21 @@ task buildWeb(type: NpmTask) {
 
     args = ['run', 'build']
 
-    inputs.dir('src')
-    inputs.files('package.json', 'package-lock.json')
+    // note that node_modules isn't cached due to potential cache misses per architecture
+    inputs.property("node.version", node.version)
+    // build per OS type to preserve behavior of packaging jar from linux
+    inputs.property("os.familyName", current().getFamilyName())
+    inputs.dir('src').withPathSensitivity(PathSensitivity.RELATIVE)
+            .withPropertyName("sources files")
+    inputs.files('package.json', 'package-lock.json', 'tsconfig.json', 'tsconfig-webpack.json',
+                 'webpack.config.ts')
+            .withPathSensitivity(PathSensitivity.RELATIVE)
+            .withPropertyName("npm configuration files")
```

Ignore if IntelliJ does not accept the indent.
```suggestion
    inputs.dir('src').withPathSensitivity(PathSensitivity.RELATIVE)
          .withPropertyName("sources files")
    inputs.files('package.json', 'package-lock.json', 'tsconfig.json', 'tsconfig-webpack.json',
                 'webpack.config.ts')
          .withPathSensitivity(PathSensitivity.RELATIVE)
          .withPropertyName("npm configuration files")
```

---

I'll have to update my intelliJ settings for groovy 😅    Let me commit your suggestion for now 👍